routevar/repo.go

```diff
@@ -0,0 +1,55 @@
+package routevar
+
+import (
+	"net/http"
+	"strings"
+
+	"sourcegraph.com/sourcegraph/go-sourcegraph/spec"
+
+	"github.com/sourcegraph/mux"
+)
+
+var (
+	// Repo captures RepoSpec strings in URL routes.
+	Repo = `{Repo:` + namedToNonCapturingGroups(spec.RepoPattern) + `}`
+
+	// RepoRev captures RepoRevSpec strings in URL routes.
+	RepoRev = Repo + `{ResolvedRev:(?:@` + namedToNonCapturingGroups(spec.ResolvedRevPattern) + `)?}`
+)
+
+// FixRepoRevVars is a mux.PostMatchFunc that cleans and normalizes
+// the route vars pertaining to a RepoRev.
+func FixRepoRevVars(req *http.Request, match *mux.RouteMatch, r *mux.Route) {
+	var keep bool
+	if rrev, present := match.Vars["ResolvedRev"]; present {
+		rrev = strings.TrimPrefix(rrev, "@")
+		if rrev != "" {
+			rev, commitID, err := spec.ParseResolvedRev(rrev)
+			if err == nil {
+				if rev != "" {
+					match.Vars["Rev"] = rev
+				}
+				if commitID != "" {
+					match.Vars["CommitID"] = commitID
+				}
+			} else {
+				keep = true
+			}
+		}
+	}
+
+	if !keep {
+		delete(match.Vars, "ResolvedRev")
```

Comment:
I think you may remove the variable `keep`, the `else` branch on L35 and delete this key by default when `err == nil`.
Also I think `rrev != ""` is not necessary because `spec.ParseResolveRev` will fail on an empty string.